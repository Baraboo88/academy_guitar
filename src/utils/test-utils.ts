import {AddCommentModel, GuitarCommentModel, GuitarModel, GuitarType} from '../types/guitar-model';
import {createApi} from '../api';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import {ReactWrapper} from 'enzyme';


export const MOCK_ERROR= 'Somme error';

export const mockGuitars: GuitarModel[] = [
  {
    id: 1,
    name: 'Честер Bass',
    vendorCode: 'SO757575',
    type: 'electric' as GuitarType,
    description: 'Замечательный малобюджетный вариант, созданный для новичков, которые отдают предпочтение мелодичным стилям. Прекрасно звучат блюз и баллады, исполненные на этой гитаре. Акустические свойства весьма высоки, в отличие от ее стоимости.',
    previewImg: 'img/guitar-1.jpg',
    stringCount: 7,
    rating: 4,
    price: 17500,
  },
  {
    id: 2,
    name: 'CURT Z300',
    vendorCode: 'TK129049',
    type: 'electric' as GuitarType,
    description: 'Эргономичность гитары и качество сборки являются, пожалуй, её главными преимуществами. Идеальное расположение в руках музыканта дополняется прочностью конструкции из клёна.',
    previewImg: 'img/guitar-8.jpg',
    stringCount: 7,
    rating: 3.5,
    price: 29500,
  },
  {
    id: 3,
    name: 'Roman LX',
    vendorCode: 'RO111111',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле класса премиум от компании CURT, собравшая в себе все самые необходимые качесва: лёгкость корпуса, прочность струн и компактный размер.',
    previewImg: 'img/guitar-6.jpg',
    stringCount: 4,
    rating: 4,
    price: 6800,
  },
  {
    id: 4,
    name: 'CURT T300',
    vendorCode: 'TK436457',
    type: 'electric' as GuitarType,
    description: 'CURT T300 - это шестиструнная электрогитара популярной линейки FPT. Модель с классическим стилем головы грифа и деки.',
    previewImg: 'img/guitar-3.jpg',
    stringCount: 6,
    rating: 5,
    price: 30000,
  },
  {
    id: 5,
    name: 'Dania Super',
    vendorCode: 'DI192138',
    type: 'acoustic' as GuitarType,
    description: 'Гитары производителя Dania пользуются популярностью у музыкантов разного уровня. В модели Super идеально сочетаются демократичная цена и качество. Корпус, выполненный из тополя позволяет корпусу долгое время сохранять свой первоначальный вид.',
    previewImg: 'img/guitar-4.jpg',
    stringCount: 7,
    rating: 4.5,
    price: 3500,
  },
  {
    id: 6,
    name: 'Честер WX',
    vendorCode: 'SO934345',
    type: 'electric' as GuitarType,
    description: 'Электрогитара с олд-скульным грифом и глянцевым корпусом для настоящих рок-музыкантов. Насыщенный звук идеально подойдет для исполнения рок-композиций.',
    previewImg: 'img/guitar-2.jpg',
    stringCount: 6,
    rating: 3.5,
    price: 15300,
  },
  {
    id: 7,
    name: 'Dania VX',
    vendorCode: 'DI082347',
    type: 'ukulele' as GuitarType,
    description: 'Укулеле - прекрасный компактный аналог классической акустической гитары. Прекрасное качество продуктов компании Dania сочетается с демократичными ценами продуктов линейки GSR.',
    previewImg: 'img/guitar-5.jpg',
    stringCount: 4,
    rating: 2.5,
    price: 2200,
  },
];
export const testInitialState = {
  currentGuitar: {
    currentGuitar: mockGuitars,
    errorMsg: '',
  },
  guitars: {
    guitars: mockGuitars,
    errorMsg: '',
  },
};

export const mockCommentToAdd: AddCommentModel = {
  advantage: 'Mock Advantage', comment: 'Mock Comment', disadvantage: 'Mock Disadvantage', guitarId: 3, rating: 3, userName: 'Some name',
};


export const mockComment: GuitarCommentModel = {id: '31b10247-adff-44a7-937e-c71af64f425b',
  userName: 'Никита',
  advantage: 'Цена.',
  disadvantage: 'Быстро приходит в негодность покрытие краской.',
  comment: 'Хорошая гитара для начинающих, сделана качественно. Лучше за эти деньги не найти.',
  rating: 5,
  createAt: new Date('2022-03-10T10:56:01.590Z'),
  guitarId: 1};

export const getTestStore = () => {

  const reducer = (state = testInitialState) => state;
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};

export const findByTestAtr = (component:ReactWrapper, attr: string) => component.find(`[data-test='${attr}']`);
