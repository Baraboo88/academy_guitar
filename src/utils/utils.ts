const STARTS_AMOUNT = 5;

export const imageAdapter = (img: string) => img.slice(img.length - 5, img.length - 4);


export const generateStarsArray = () => {
  const stars = [];
  for (let i = 1; i <= STARTS_AMOUNT; i++) {
    stars.push(i);
  }
  return stars;
};

export const getCyrillicRating = (rating: number) => {
  if(rating> 0 && rating < 3){
    return 'Плохо';
  } else if (rating >= 3 && rating < 4){
    return 'Удовлетворительно';
  } else if(rating >= 4 && rating < 5){
    return 'Хорошо';
  } else {
    return 'Отлично';
  }
};
