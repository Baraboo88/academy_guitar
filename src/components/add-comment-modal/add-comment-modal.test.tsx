
import * as Enzyme from 'enzyme';

import { mount } from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';

import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import AddCommentModal from './add-comment-modal';
import {getTestStore, mockCommentToAdd, mockGuitars} from '../../utils/test-utils';


Enzyme.configure({adapter: new EnzymeReactAdapter()});


it('AddCommentModal successfully rendered', () => {
  const mockCloseModalHandler = jest.fn();
  const mockSetUserNameHandler = jest.fn();
  const mockSetAdvantageHandler = jest.fn();
  const mockSetDisadvantageHandler = jest.fn();
  const mockSetCommentHandler = jest.fn();
  const mockSetRatingHandler = jest.fn();
  const mockOnSubmitHandler = jest.fn();
  const tree = mount(
    <Provider store={getTestStore()}>
      <AddCommentModal
        guitar={mockGuitars[0]}
        advantage={mockCommentToAdd.advantage}
        comment={mockCommentToAdd.comment}
        disadvantage={mockCommentToAdd.disadvantage}
        rating={mockCommentToAdd.rating}
        userName={mockCommentToAdd.userName}
        onCloseModalHandler={mockCloseModalHandler}
        setUserNameHandler={mockSetUserNameHandler}
        setAdvantageHandler={mockSetAdvantageHandler}
        setDisadvantageHandler={mockSetDisadvantageHandler}
        setCommentHandler={mockSetCommentHandler}
        setRatingHandler={mockSetRatingHandler}
        onSubmitHandler={mockOnSubmitHandler}

      />
    </Provider>,
  );
  expect(toJson(tree, {mode: 'deep'})).toMatchSnapshot();
});
