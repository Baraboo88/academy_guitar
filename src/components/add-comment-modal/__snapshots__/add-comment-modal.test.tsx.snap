// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCommentModal successfully rendered 1`] = `
<div
  className="modal is-active modal--review modal-for-ui-kit"
>
  <div
    className="modal__wrapper"
  >
    <div
      className="modal__overlay"
      data-close-modal={true}
      data-test="test-close-modal"
      onClick={[MockFunction]}
    />
    <div
      className="modal__content"
    >
      <h2
        className="modal__header modal__header--review title title--medium"
      >
        Оставить отзыв
      </h2>
      <h3
        className="modal__product-name title title--medium-20 title--uppercase"
      >
        Честер Bass
      </h3>
      <form
        className="form-review"
        data-test="test-submit"
        onSubmit={[Function]}
      >
        <div
          className="form-review__wrapper"
        >
          <div
            className="form-review__name-wrapper"
          >
            <label
              className="form-review__label form-review__label--required"
              htmlFor="user-name"
            >
              Ваше Имя
            </label>
            <input
              autoComplete="off"
              autoFocus={true}
              className="form-review__input form-review__input--name"
              data-test="test-user-name"
              id="user-name"
              onChange={[Function]}
              type="text"
              value="Some name"
            />
          </div>
          <div>
            <span
              className="form-review__label form-review__label--required"
            >
              Ваша Оценка
            </span>
            <div
              className="rate rate--reverse"
            >
              <input
                checked={false}
                className="visually-hidden"
                data-test="test-rating"
                id="star-5"
                name="rate"
                onChange={[Function]}
                onKeyDown={[Function]}
                type="radio"
                value={5}
              />
              <label
                className="rate__label"
                htmlFor="star-5"
                title="Отлично"
              />
              <input
                checked={false}
                className="visually-hidden"
                data-test="test-rating"
                id="star-4"
                name="rate"
                onChange={[Function]}
                onKeyDown={[Function]}
                type="radio"
                value={4}
              />
              <label
                className="rate__label"
                htmlFor="star-4"
                title="Хорошо"
              />
              <input
                checked={true}
                className="visually-hidden"
                data-test="test-rating"
                id="star-3"
                name="rate"
                onChange={[Function]}
                onKeyDown={[Function]}
                type="radio"
                value={3}
              />
              <label
                className="rate__label"
                htmlFor="star-3"
                title="Нормально"
              />
              <input
                checked={false}
                className="visually-hidden"
                data-test="test-rating"
                id="star-2"
                name="rate"
                onChange={[Function]}
                onKeyDown={[Function]}
                type="radio"
                value={2}
              />
              <label
                className="rate__label"
                htmlFor="star-2"
                title="Плохо"
              />
              <input
                checked={false}
                className="visually-hidden"
                data-test="test-rating"
                id="star-1"
                name="rate"
                onChange={[Function]}
                onKeyDown={[Function]}
                type="radio"
                value={1}
              />
              <label
                className="rate__label"
                htmlFor="star-1"
                title="Ужасно"
              />
            </div>
          </div>
        </div>
        <label
          className="form-review__label form-review__label--required"
          htmlFor="adv"
        >
          Достоинства
        </label>
        <input
          autoComplete="off"
          className="form-review__input"
          data-test="test-advantage"
          id="adv"
          onChange={[Function]}
          type="text"
          value="Mock Advantage"
        />
        <label
          className="form-review__label form-review__label--required"
          htmlFor="disadv"
        >
          Недостатки
        </label>
        <input
          autoComplete="off"
          className="form-review__input"
          data-test="test-disadvantage"
          id="disadv"
          onChange={[Function]}
          type="text"
          value="Mock Disadvantage"
        />
        <label
          className="form-review__label form-review__label--required"
          htmlFor="comment"
        >
          Комментарий
        </label>
        <textarea
          autoComplete="off"
          className="form-review__input form-review__input--textarea"
          data-test="test-comment"
          id="comment"
          onChange={[Function]}
          rows={10}
          value="Mock Comment"
        />
        <button
          className="button button--medium-20 form-review__button"
          type="submit"
        >
          Отправить отзыв
        </button>
      </form>
      <button
        aria-label="Закрыть"
        className="modal__close-btn button-cross"
        data-test="test-close-modal"
        onClick={[MockFunction]}
        type="button"
      >
        <span
          className="button-cross__icon"
        />
        <span
          className="modal__close-btn-interactive-area"
        />
      </button>
    </div>
  </div>
</div>
`;
